gateway/internal/application/service/user_service.go

```diff
@@ -0,0 +1,101 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	userdto "github.com/Barbod-Biometrics/Backend/gateway/internal/application/dto/user"
+	"github.com/Barbod-Biometrics/Backend/gateway/internal/application/usecase"
+	"github.com/Barbod-Biometrics/Backend/gateway/internal/domain/entity"
+	"github.com/Barbod-Biometrics/Backend/gateway/internal/domain/repository"
+)
+
+var (
+	ErrUserNotFound      = errors.New("user not found")
+	ErrEmailAlreadyExist = errors.New("email already exists")
+	ErrPhoneAlreadyExist = errors.New("phone number already exists")
+	ErrInvalidUpdateData = errors.New("no valid data to update")
+)
+
+type UserService struct {
+	UserRepo repository.UserRepository
+}
+
+func NewUserService(userRepo repository.UserRepository) *UserService {
+	return &UserService{
+		UserRepo: userRepo,
+	}
+}
+
+var _ usecase.UserService = (*UserService)(nil)
+
+func (s *UserService) GetUserByID(ctx context.Context, userID uint64) (*entity.User, error) {
```

Comment:
Hmm, when I get user info via the GET method, the response contains the is_admin flag. which I don't think we should return in that api response.

```suggestion
func (s *UserService) GetUserByID(ctx context.Context, userID uint64) (*userdto.UserInfoResponse, error) {
```